cartoframes/context.py

```diff
@@ -724,8 +727,21 @@ def safe_quotes(text, escape_single_quotes=False):
                      width=size[0],
                      height=size[1],
                      img_html=img_html)
+            return IPython.display.HTML(html)
+        else:
+            try:
```

Comment:
Instead of putting this here, let's move it to the beginning with a global bool on whether matplotlib can be used. Something like this:

```python
try:
    import matplotlib.image as mpi
    import matplotlib.pyplot as plt
    HAS_MATPLOTLIB = True
except ImportError:
    HAS_MATPLOTLIB = False
```

Then we can use that bool in other places as we need it (e.g., legend creation for interactive maps)